src/ray/util/filesystem.cc

```diff
@@ -0,0 +1,47 @@
+#include "ray/util/filesystem.h"
+
+#include <stdlib.h>
+
+#include "ray/util/logging.h"
+
+#ifdef _WIN32
+#include <Windows.h>
+#endif
+
+namespace ray {
+
+std::string GetRayTempDir() { return JoinPaths(GetUserTempDir(), "ray"); }
+
+std::string GetUserTempDir() {
+  std::string result;
+#if defined(__APPLE__) || defined(__linux__)
+  // Prefer the hard-coded path for now, for compatibility.
+  result = "/tmp";
+#elif defined(_WIN32)
+  result.resize(1 << 8);
+  DWORD n = GetTempPath(static_cast<DWORD>(result.size()), &*result.begin());
+  if (n > result.size()) {
+    result.resize(n);
+    n = GetTempPath(static_cast<DWORD>(result.size()), &*result.begin());
+  }
+  result.resize(0 < n && n <= result.size() ? static_cast<size_t>(n) : 0);
+#else
+  const char *candidates[] = {"TMPDIR", "TMP", "TEMP", "TEMPDIR"};
```

Comment:
Ah, this block also needs to be changed, why don't you move it into the windows ifdef?

We need to make sure the behavior doesn't change on the other platforms like linux and macOS (or if we do, it needs a longer discussion).

Reply:
Ah, I misunderstood, this is only run on other platforms.